File: src/main/java/io/github/robertomike/utils/ResolverPathUtil.java
package io.github.robertomike.utils;

import org.springframework.web.context.request.NativeWebRequest;
import org.springframework.web.servlet.HandlerMapping;

import java.util.Map;
import java.util.Optional;

/**
 * Class for resolve path variable
 */
public class ResolverPathUtil {
    @SuppressWarnings("unchecked")
    public static Optional<String> resolveVariable(NativeWebRequest request, String variable) {
        Map<String, String> uriTemplateVars = (Map<String, String>) request.getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE, 0);
        return Optional.ofNullable(uriTemplateVars != null ? uriTemplateVars.get(variable) : null);
    }
}

File: src/main/java/io/github/robertomike/inject_model/utils/ResolverPathUtil.kt
package io.github.robertomike.inject_model.utils

import org.springframework.web.context.request.NativeWebRequest
import org.springframework.web.servlet.HandlerMapping

/**
 * Class for resolve path variable
 */
class ResolverPathUtil {
    companion object {
        /**
         * Resolves a variable from the request's URI template variables.
         *
         * @param request the current request
         * @param variable the name of the variable to resolve
         * @return the value of the variable, or null if not found
         */
        @JvmStatic
        fun resolveVariable(request: NativeWebRequest, variable: String): String? {
            val uriTemplateVars =
                request.getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE, 0) as Map<String, String>?
            return uriTemplateVars?.get(variable)
        }
    }
}